graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/shared/GenerateKotlinCode.kt

```diff
@@ -31,56 +31,80 @@ fun generateKotlinCode(
     config: CodeGenConfig,
     typeUtils: KotlinTypeUtils
 ): CodeBlock {
-    return when (value) {
-        is BooleanValue -> CodeBlock.of("%L", value.isValue)
-        is IntValue -> CodeBlock.of("%L", value.value)
-        is StringValue -> {
-            val localeValueOverride = checkAndGetLocaleValue(value, type)
-            if (localeValueOverride != null) CodeBlock.of("%L", localeValueOverride)
-            else CodeBlock.of("%S", value.value)
-        }
-        is FloatValue -> CodeBlock.of("%L", value.value)
-        is EnumValue -> CodeBlock.of("%M", MemberName(type.className, value.name))
-        is ArrayValue ->
-            if (value.values.isEmpty()) CodeBlock.of("emptyList()")
-            else CodeBlock.of(
-                "listOf(%L)",
-                value.values.joinToString { v -> generateKotlinCode(v, type, inputTypeDefinitions, config, typeUtils).toString() }
-            )
-
-        is ObjectValue -> {
-            val inputObjectDefinition = inputTypeDefinitions.first {
-                val expectedCanonicalClassName = config.typeMapping[it.name] ?: "${config.packageNameTypes}.${it.name}"
-                expectedCanonicalClassName == type.className.canonicalName
+    return checkAndGetLocaleCodeBlock(value, type)
+        ?: checkAndGetBigDecimalCodeBlock(value, type)
+        ?: when (value) {
+            is BooleanValue -> CodeBlock.of("%L", value.isValue)
+            is IntValue -> CodeBlock.of("%L", value.value)
+            is StringValue -> {
+                val localeValueOverride = checkAndGetLocaleCodeBlock(value, type)
+                if (localeValueOverride != null) CodeBlock.of("%L", localeValueOverride)
```

Comment:
I would think this would need to change, as you changed the above method to return a `CodeBlock`.

Reply:
Oh actually looks like you probably can get rid of most of this and just simplify to
`is StringValue -> CodeBlock.of("%S", value.value)`

Reply:
Yup, missed that one. 👍